tests/test_recharge_start_date.py

```diff
@@ -0,0 +1,92 @@
+
+        # run initial sync
+        record_count_by_stream_1 = self.run_and_verify_sync(conn_id_1)
+        synced_records_1 = runner.get_records_from_target_output()
+
+        ##########################################################################
+        ### Update START DATE Between Syncs
+        ##########################################################################
+
+        print("REPLICATION START DATE CHANGE: {} ===>>> {} ".format(self.start_date, self.start_date_2))
+        self.start_date = self.start_date_2
+
+        ##########################################################################
+        ### Second Sync
+        ##########################################################################
+
+        # create a new connection with the new start_date
+        conn_id_2 = connections.ensure_connection(self, original_properties=False)
+
+        # run check mode
+        found_catalogs_2 = self.run_and_verify_check_mode(conn_id_2)
+
+        # table and field selection
+        test_catalogs_2_all_fields = [catalog for catalog in found_catalogs_2
+                                      if catalog.get('tap_stream_id') in expected_streams]
+        self.perform_and_verify_table_and_field_selection(conn_id_2, test_catalogs_2_all_fields, select_all_fields=True)
+
+        # run sync
+        record_count_by_stream_2 = self.run_and_verify_sync(conn_id_2)
+        synced_records_2 = runner.get_records_from_target_output()
+
+        for stream in expected_streams:
+            with self.subTest(stream=stream):
+                # expected values
+                expected_primary_keys = self.expected_primary_keys()[stream]
+
+                # collect information for assertions from syncs 1 & 2 base on expected values
+                record_count_sync_1 = record_count_by_stream_1.get(stream, 0)
+                record_count_sync_2 = record_count_by_stream_2.get(stream, 0)
+                primary_keys_list_1 = [tuple(message.get('data').get(expected_pk) for expected_pk in expected_primary_keys)
+                                       for message in synced_records_1.get(stream, {}).get('messages', {})
+                                       if message.get('action') == 'upsert']
+                primary_keys_list_2 = [tuple(message.get('data').get(expected_pk) for expected_pk in expected_primary_keys)
+                                       for message in synced_records_2.get(stream, {}).get('messages', {})
+                                       if message.get('action') == 'upsert']
+                primary_keys_sync_1 = set(primary_keys_list_1)
+                primary_keys_sync_2 = set(primary_keys_list_2)
+
+                # Verify that the 2nd sync with a later start date replicates the same number of
+                # records as the 1st sync.
+                self.assertEqual(record_count_sync_2, record_count_sync_1)
```

Comment:
The standard expectation is that all streams do obey the start date except when this conflicts with the business logic of the integration. I would like to check with product to make sure this is true for this integration.

Reply:
The 2 cases where getting records prior to the start date is acceptable are:
(1) Technical limitations like not being able to query on datetime value in order to filter records
(2) It makes more sense with how the data is structured and the underlying business logic to grab all records on an initial sync. 

Could you please explain if either of these are the case for all streams. If this is not the case the tap behavior should change to meet the standard expectation.

Reply:
@kspeer825 I think I know what happened here, I was looking the [example](https://github.com/singer-io/tap-facebook/blob/09e173b45a27bd8b6cb91f28ca9ad3994349b25b/tests/test_facebook_start_date.py#L107) for the [start_date test](https://github.com/stitchdata/tap-tester/blob/master/reference/expectations/saas_taps.md#example-tests) and it is uses a specific check to the `tap-facebook` implementation. So only the `FULL_TABLE` relevant assertions were added and due to the data and start_date the tests were passing.

I've adjusted the assertions to be more relevant for `INCREMENTAL` streams and the data. This should cover this scenario more accurately now.

Reply:
@loeakaodas I see, this is really helpful feedback. The tap-tester expectations are still a work in progress, so if you notice discrepancies between examples and what is mentioned in reviews please let me know. I will try to find a better example to link in that doc.